Homeworks/Homework_5/transfers_script.py

```diff
@@ -0,0 +1,347 @@
+import csv
+import sqlite3
+from datetime import datetime, timezone
+from Homeworks.Homework_5.validations import validate_bank_data, validate_user_data, validate_account_data
+from Homeworks.Homework_5.utils import establish_db_connection, get_logger, get_currency, SELECT_COMMAND, UPDATE_COMMAND
+
+logger = get_logger()
+
+
+def add_data(cursor, input_data, table_name, params_insert, message_return, vld_fnc):
+    """
+    Add data to a database table using the provided cursor.
+    This function inserts data into a specified database table using the given cursor.
+
+    :param cursor: (sqlite3.Cursor) The cursor object to execute SQL commands.
+    :param input_data: (list) List of data to be inserted into the table.
+    :param table_name: (str) The name of the target database table.
+    :param params_insert: (str) Comma-separated string of column names for insertion.
+    :param message_return: (str) Message to be returned upon successful insertion.
+    :param vld_fnc: (callable) Validation function to validate input_data.
+
+    :return: (str) The message_return upon successful data insertion.
+
+    :raise Exception: If validation fails for any of the input_data.
+    """
+    for data in input_data:
+        vld_fnc(data)
+        question_marks = ('?,' * len(params_insert.split(',')))[:-1]
+        cursor.execute(f'INSERT INTO {table_name} ({params_insert}) VALUES ({question_marks})', data)
+    return message_return
+
+
+@establish_db_connection
+def add_bank(cursor, *bank_data):
+    """
+    Add banks to the database.
+
+    :param cursor: (sqlite3.Cursor) The cursor to interact with the database.
+    :param bank_data: (list) A list of bank names to be added.
+
+    :return: (str) The status of the operation.
+    """
+    return add_data(cursor, bank_data, 'Bank', 'name', 'Banks added successfully.', validate_bank_data)
+
+
+def parse_user_full_name(user_data):
+    """
+    Parse user full name data into a formatted list.
+    This function takes a list of user data where each element contains the first name,
+    last name, and additional information. It parses this data and returns a list
+    of tuples containing formatted first name, last name, and additional information.
+
+    :param user_data: (list) List of user data where each element contains first name,
+                             last name, and additional information.
+
+    :return: (list) A list of tuples with formatted user information.
+    """
+    return [(*i[0].split(), *i[1:]) for i in user_data]
```

Comment:
didn't find this, please point me if you have such code:
![image](https://github.com/MykytaKa/PyCourse/assets/43804299/8e6a28e2-404e-49ec-bf3b-a43ec6859fe0)